plasmapy/formulary/lengths.py

```diff
@@ -225,12 +235,21 @@ def gyroradius(
 
     isfinite_T = np.isfinite(T)
     isfinite_Vperp = np.isfinite(Vperp)
+    isfinite_lorentzfactor = np.isfinite(lorentzfactor)
 
     # check 1: ensure either Vperp or T invalid, keeping in mind that
     # the underlying values of the astropy quantity may be numpy arrays
-    if np.any(np.logical_and(isfinite_Vperp, isfinite_T)):
-        raise ValueError(
-            "Must give Vperp or T, but not both, as arguments to gyroradius"
+    if np.any(np.logical_not(np.logical_xor(isfinite_Vperp, isfinite_T))):
+        # if neither are provided calc Vperp from lorentz factor but but you need lorentz factor to have no undefined elements
+        if np.all(isfinite_lorentzfactor):
+            Vperp = RelativisticBody(particle, lorentzfactor).velocity
+        else:
+            raise ValueError(
+                "Must give one of Vperp or T, but not both, as arguments to gyroradius, and if not then provide the lorentzfactor"
```

Comment:
Probably clearer this way?
```suggestion
                "Must give either the lorentzfactor or one of: Vperp or T"
```